### PR TITLE
RDM-12591: Create Case issue with Security Classification

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-12591

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/ClassifiedCreateEventOperation.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public CaseDetails createCaseEvent(String caseReference,
         final CaseDetails caseDetails = createEventOperation.createCaseEvent(caseReference,
                                                                            content);
         return Optional.ofNullable(caseDetails)
-                       .flatMap(classificationService::applyClassification)
+                       .flatMap(details -> classificationService.applyClassification(details, true))
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/ClassifiedCreateEventOperation.java:32]: We're passing `create` as `true` here, but the case already exists when creating an event - is this correct?

File: src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationService.java [hunk 2]
```diff
@@ -35,9 +35,14 @@ public AttributeBasedSecurityClassificationService(@Qualifier(CachedUserReposito
 
 
     @Override
-    public Optional<SecurityClassification> getUserClassification(CaseDetails caseDetails) {
-        Set<AccessProfile> accessProfiles = caseDataAccessControl
-            .generateAccessProfilesByCaseReference(caseDetails.getReferenceAsString());
+    public Optional<SecurityClassification> getUserClassification(CaseDetails caseDetails, boolean create) {
+        Set<AccessProfile> accessProfiles;
+        if (create) {
+            accessProfiles = caseDataAccessControl.generateAccessProfilesByCaseTypeId(caseDetails.getCaseTypeId());
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationService.java:41]: Add tests for this please

Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationService.java:41]: Test coverage missing for the new flow

**Review by danlysiak — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/ClassifiedCreateEventOperation.java (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/ClassifiedCreateEventOperation.java]: remove this now

File: src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationService.java (hunk 2 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationService.java:41]: added test coverage report

**Pull request merged**